File: library/src/main/java/org/wbing/refresh/WRefreshSingleListFragment.java
package org.wbing.refresh;

import android.databinding.ViewDataBinding;

import org.wbing.view.list.WAdapter;

/**
 * @author wangbing
 * @date 2018/8/23
 */
public abstract class WRefreshSingleListFragment<Data, Binding extends ViewDataBinding>
        extends WRefreshListFragment<Data, Binding> {

    private WAdapter<Data, Binding> adapter;


    @Override
    public WAdapter<Data, Binding> adapter() {
        if (adapter == null) {
            adapter = new WAdapter.SimpleAdapter<>(variableId(), holderLayout());
        }
        return adapter;
    }

    @Override
    public void recycle() {
        adapter.removeAll();
        adapter = null;
    }

    protected abstract int variableId();

    protected abstract int holderLayout();
}

File: library/src/main/java/org/wbing/refresh/WRefreshSingleListFragment.kt
package org.wbing.refresh

import android.databinding.ViewDataBinding
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import org.wbing.view.list.WAdapter
import org.wbing.view.list.WMultiAdapter
import org.wbing.view.list.WRecyclerView

/**
 * @author wangbing
 * @date 2018/8/23
 */
abstract class WRefreshSingleListFragment<Data, Binding : ViewDataBinding> : WRefreshAdapterListFragment<Data, Binding>() {

    private var defaultAdapter: WAdapter<Data, Binding> = WAdapter.SimpleAdapter<Data, Binding>(this.variableId(), this.holderLayout())
    private var defaultLayoutManager: RecyclerView.LayoutManager = LinearLayoutManager(context)

    override fun initRecyclerView(view: WRecyclerView) {
        view.layoutManager = layoutManager()
        view.adapter = adapter()
    }

    override fun adapter(): WAdapter<Data, Binding> {
        return defaultAdapter
    }

    override fun layoutManager(): RecyclerView.LayoutManager {
        return defaultLayoutManager
    }

    protected abstract fun variableId(): Int
    protected abstract fun holderLayout(): Int
}
